app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+
+    if user&.authenticate(params[:session][:password])
+      # Log the user in and redirect to the user's show page.
+      cookies.permanent[:user_id] = user.id
+      cookies.permanent[:remember_token] = user.remember_token
+      flash[:success] = 'Welcome!'
+    else
+      # Create an error message.
+      flash[:error] = 'Wrong combination'
+    end
+    redirect_to signup_path
+  end
+
+  def destroy
```

Comment:
This method should set the` @current_user to nil` [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/session_delete_requirement.png)